MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs

```diff
@@ -0,0 +1,217 @@
+                out value))
+        {
+            return true;
+        }
+
+        logger.LogWarning("Invalid date: {Date}", text);
+        return false;
+    }
+
+    /// <summary>
+    /// Parses a <see cref="DateTime"/> from the current node.
+    /// </summary>
+    /// <param name="reader">The <see cref="XmlReader"/>.</param>
+    /// <param name="formatString">The date format string.</param>
+    /// <param name="value">The parsed <see cref="DateTime"/>.</param>
+    /// <returns>A value indicating whether the parsing succeeded.</returns>
+    public static bool TryReadDateTimeExact(this XmlReader reader, string formatString, out DateTime value)
+    {
+        ArgumentNullException.ThrowIfNull(reader);
+        ArgumentNullException.ThrowIfNull(formatString);
+
+        return DateTime.TryParseExact(
+            reader.ReadElementContentAsString(),
+            formatString,
+            CultureInfo.InvariantCulture,
+            DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal,
+            out value);
+    }
+
+    /// <summary>
+    /// Parses a <see cref="PersonInfo"/> from the xml node.
+    /// </summary>
+    /// <param name="reader">The <see cref="XmlReader"/>.</param>
+    /// <returns>A <see cref="PersonInfo"/>, or <c>null</c> if none is found.</returns>
+    public static PersonInfo? GetPersonFromXmlNode(this XmlReader reader)
+    {
+        ArgumentNullException.ThrowIfNull(reader);
+
+        if (reader.IsEmptyElement)
+        {
+            reader.Read();
+            return null;
+        }
+
+        var name = string.Empty;
+        var type = PersonKind.Actor;  // If type is not specified assume actor
+        var role = string.Empty;
+        int? sortOrder = null;
+        string? imageUrl = null;
+
+        using var subtree = reader.ReadSubtree();
+        subtree.MoveToContent();
+        subtree.Read();
+
+        while (subtree is { EOF: false, ReadState: ReadState.Interactive })
+        {
+            if (subtree.NodeType != XmlNodeType.Element)
+            {
+                subtree.Read();
+                continue;
+            }
+
+            switch (subtree.Name)
+            {
+                case "name":
+                case "Name":
+                    name = subtree.ReadElementContentAsString();
+                    break;
+                case "role":
+                case "Role":
+                    var roleValue = subtree.ReadElementContentAsString();
+                    if (!string.IsNullOrWhiteSpace(roleValue))
+                    {
+                        role = roleValue;
+                    }
+
+                    break;
+                case "type":
+                case "Type":
+                    Enum.TryParse(subtree.ReadElementContentAsString(), true, out type);
+                    break;
+                case "order":
+                case "sortorder":
+                case "SortOrder":
+                    if (int.TryParse(
```

Comment:
shouldn't it be using the TryReadInt here?

Reply:
Yes, I've updated this method to use the other extension methods